### PR TITLE
Add test for MutableMapping methods on object_maps and fix bug

### PR DESCRIPTION
**Pull request by hoodmane**
This adds tests of all the MutableMapping methods for objects returned by `as_object_map`. They all worked but I was accidentally adding only the non mutating methods. I also added `JsMap` to `pyodide.ffi` and fixed it so that `isinstance` works correctly and mypy agrees with the usage.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry

### CODE REVIEW

File: src/py/_pyodide/_core_docs.py [hunk 1]
```diff
@@ -441,6 +447,38 @@ def reverse(self) -> None:
 
 
 class JsMap(JsProxy):
```
Comment (hoodmane) [inline src/py/_pyodide/_core_docs.py:449]: One question: perhaps we should have a `JsMap` and a `JsMutableMap`?

Comment (ryanking13) [reply src/py/_pyodide/_core_docs.py:449]: I think it is a good idea.

File: src/py/_pyodide/_core_docs.py [hunk 2]
```diff
@@ -441,7 +448,35 @@ def reverse(self) -> None:
...
+
+    _js_type_flags = ["HAS_GET | HAS_LENGTH | IS_ITERABLE", "IS_OBJECT_MAP"]
+
+    def __getitem__(self, idx: Any) -> Any:
+        return None
+
+    def __setitem__(self, idx: Any, value: Any) -> None:
+        pass
+
+    def __delitem__(self, idx: Any) -> None:
```
Comment (ryanking13) [inline src/py/_pyodide/_core_docs.py:468]: Isn't `__setitem__` and `__delitem__` for `JsMutableMap`?

**Review by ryanking13 — APPROVED** [review]
Thanks, @hoodmane!

File: src/py/_pyodide/_core_docs.py (hunk 2 above)
Comment (hoodmane) [reply src/py/_pyodide/_core_docs.py]: Yes good point.

**Pull request merged**